### PR TITLE
fix(rpc): always include withdrawals in RPC

### PR DESCRIPTION
**Pull request by mattsse**
this fixes the `debug_getRawBlock` hive test.

In RPC, withdrawals are always included, no withdrawals are empty list.

### CODE REVIEW

**Review by onbjerg — APPROVED** [review]
lgtm pending merge conflicts

File: crates/rpc/rpc/src/debug.rs [hunk 1]
```diff
@@ -155,7 +155,11 @@ where
         let block = self.client.block(block_id).to_rpc_result()?;
 
         let mut res = Vec::new();
-        if let Some(block) = block {
+        if let Some(mut block) = block {
+            // In RPC withdrawals are always present
+            if block.withdrawals.is_none() {
+                block.withdrawals = Some(vec![]);
+            }
```
Comment (gakonst) [inline crates/rpc/rpc/src/debug.rs:162]: I see. A bit meh footgun here. Should it be `withdrawals: Vec<_>`, instead of `withdrawals: Option<Vec<_>>`?

**Review by gakonst — APPROVED** [review]

File: crates/rpc/rpc/src/debug.rs (hunk 1 above)
Comment (Rjected) [reply crates/rpc/rpc/src/debug.rs:162]: Wouldn't doing that break the RLP impl because of how we use `rlp(trailing)` and the option?

Comment (mattsse) [reply crates/rpc/rpc/src/debug.rs:162]: this more of a quirk of geth, wich does not have options so empty is the same as missing so missing is also encoded as trailing `c0`

**Pull request merged**